src/WooCommerce/HttpClient/HttpClient.php

```diff
@@ -324,6 +331,10 @@ protected function setDefaultCurlSettings()
         \curl_setopt($this->ch, CURLOPT_RETURNTRANSFER, true);
         \curl_setopt($this->ch, CURLOPT_HTTPHEADER, $this->request->getRawHeaders());
         \curl_setopt($this->ch, CURLOPT_URL, $this->request->getUrl());
+
+        foreach ($this->customCurlOptions as $customCurlOptionKey => $customCurlOptionValue) {
```

Comment:
@redelschaap it will not work if you use Client because customCurlOptions are missing in constructor.

> public function __construct($url, $consumerKey, $consumerSecret, $options = [])

The easiest way to implement it using $options and have a foreach here by prefix
[
    "curl_*" => "value"
]

Reply:
Well, I have also added a method to set the custom CURL options, so it would need to be used (for example) like this:

```php
$client = new Client($url, $consumerKey, $consumerSecret, $options);

$client->http->setCustomCurlOptions([
    CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_2,
]);

$products = $client->get('products');
```